keps/sig-release/20190118-creating-packages.md

```diff
@@ -0,0 +1,417 @@
+   - must take the following inputs:
+     - the package configuration
+     - the pre-compiled binarie(s) 
+     - the version the binaries are compiled from
+   - might take as input:
+     - the platform(s) to package for (default: [all supported platforms](#plat-list))
+   - must create the following outputs:
+     - one package per platform (`linux/386`, `darwin/amd64`, ...) and package type (`rpm`, `deb`)
+
+   Example run:
+   ```bash
+   docker build -t k8s-pkg-builder build/pkgs
+   docker run \
+     -v ${PWD}/_output/dockerized/bin:/in \
+     -v ${PWD}/_output/pkgs/nightly:/out \
+     -v ${PWD}/build/pkgs/kubeadm:/conf \
+     -e KUBE_BUILD_VERSION='1.12.6~beta.0.1+12187918e930d4+dirty' \
+     -e KUBE_BUILD_PLATFORMS='linux/amd64 linux/386' \
+     k8s-pkg-builder
+   ```
+
+- [ ] Create all package configurations for all currently published
+  [packages](#pkg-list) and [platforms](#plat-list)
+
+- [ ] The output of the above tool (a directory tree, holding a set of
+  generated packages) ideally is in a format that can be uploaded to a GCS
+  bucket which can be used as a debian package repo / yum repo.
+
+  If that is not feasible there needs to be a tool that can re-arrange that
+  directory and prepare it for upload. This might also give the opportunity to
+  run certain tools to prepare the directory, e.g. [`dpkg-scanpacakges`,
+  `debpool` or other).
+
+- [ ] Create one tool that does "everything" that needs to be done in the
+  release and can be called by [`anago`][anago].
+
+  Without any special configuration is should do whatever is needed so that it
+  can be called from within [`anago`][anago], however there should be options
+  to specify which components to package, which components to package, get the
+  artefacts by compiling them or pulling them from a staging bucket, ...
+  
+  This might be seen as the
+  "orchestrator tool" and has the following responsibilities:
+  1. determinate the version to be packaged, based on the current git checkout
+     ok `k/k` (a version like in `bazel-genfiles/build/version`)
+  1. do a build (`build/run.sh make $COMPONENT` or such) or download all needed
+     artefacts from from a staging bucket
+  1. run the tool to prepare packages for upload
+  1. upload the packages
+
+
+[anago]: https://github.com/kubernetes/release/blob/master/anago
+
+<!--
+### User Stories [optional]
+
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of the system.
+The goal here is to make this feel real for users without getting bogged down.
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they releate.
+-->
+
+### Risks and Mitigations
```

Comment:
possibly not to be present in this list,

but tech dept of the tool we plan to write and maintain seems like a concern to me.
i think we should use go and write it in a object oriented manner.
-1 on bash.